### ISSUE TITLE
Question: In 2.0.0, is VueTypes.string.def("") supposed to work still?

### ISSUE THREAD

**Issue by attekemppila** [dwightjack/vue-types#68]
In 2.0.0 you can import individual validators: `import { string } from "vue-types"`

But is the old style supposed to work still?

```
import VueTypes from "vue-types";

props: { foo: VueTypes.string.def("") }
```

It gives me an error: `Uncaught TypeError: _vueTypes2.default.string.def is not a function`

Basically `VueTypes.string` is a function, and `VueTypes.string.def` is undefined.

Usage example implies that old style should still work: https://dwightjack.github.io/vue-types/#usage-example

**Comment by dwightjack** [comment]
Hello, 

I have updated the demo sandbox using namespaced validators and it seems to work as expected: https://codesandbox.io/s/vue-types-2-demo-rbrdh?file=/src/components/ListItem.vue

I am not sure why you are getting that error. Could you share the component's code? If I should guess, I'd say that's what you'd get by importing the library as:

```js
import * as VueTypes from 'vue-types'
``` 

Maybe some bundler plugin / compiler config? 🤔

**Comment by attekemppila** [comment]
I'm using browserify (+babelify 8 (= babel 6, can't upgrade easily because of reasons)).

Maybe module glue from those interferes somehow. I don't know.

No need to dig into this deeper on my account. I have been meaning to move away from browserify anyway in the near future.

For now, I'll just stay in 1.x or move to use individual validator syntax (`import { string } from "vue-types"`) which seems to work ok.

Thanks for quick answer.

**Comment by dwightjack** [comment]
Thanks for the details.

I might have found the problem. Browserify uses commonjs as module standard instead of esm (used by default by webpack and rollup), and seems that microbundle (which I'm using for bundling the library) has some issues with [commonjs and default exports](https://github.com/developit/microbundle/issues/306).

I will investigate if the issue is fixable. In the meantime I think it's pretty safe to stick to 1.x.

I'll keep you updated.

**Comment by nolde** [comment]
I am getting the same thing when running my jest unit tests.
```
 TypeError: _vueTypes.default.number.def is not a function
TypeError: _vueTypes.default.bool.def is not a function
```
... not sure why though.

Using vue cli 3 packing, little customisation. Using `import VueTypes from "vue-types"` format.

Also, you should probably mention on your notes that `boolean` was updated to `bool`.

**Comment by dwightjack** [comment]
@nolde This should confirm my guess: Jest uses commonjs to load modules so it's loading a version of the library affected by the issue I linked above (https://github.com/dwightjack/vue-types/issues/68#issuecomment-672931304).

I published the `2.0.1-rc.1` version as beta. It should fix the issue. To install run `npm i vue-types@2.0.1-rc.1 -S`

If that works I'll publish `2.0.1` as stable.

Sorry for the issue.

**Comment by attekemppila** [comment]
2.0.1-rc.1 seems to work in my case. Both import methods: `import VueTypes, { string } from "vue-types"`

**Comment by nolde** [comment]
Tests passed after installing `rc.1`!

**Comment by dwightjack** [comment]
@attekemppila @nolde Thanks for testing it out.

I published the fix in `2.0.1`.
